pkgs/_test_macros/lib/json_codable.dart

```diff
@@ -49,7 +49,28 @@ class JsonCodableImplementation implements Macro {
     final target = request.target;
     final model = await host.query(Query(target: target));
     final clazz = model.uris[target.uri]!.scopes[target.name]!;
-    // TODO(davidmorgan): check for super `fromJson`.
+
+    MacroScope.current.addModel(model);
+    final superclassName = MacroScope.current.typeSystem.supertypeOf(target);
+
+    var superclassHasFromJson = false;
```

Comment:
nit: I think it would help readability to extract the toJson/fromJson parts into separate functions